cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -292,6 +293,12 @@ protected void startUp() throws Exception {
         Collections.sort(jarFiles);
         classpath.addAll(jarFiles);
         classpath.add("job.jar/classes");
+        // Add extra jars set in cConf
+        for (File extraJar : CConfigurationUtil.getExtraJars(cConf)) {
+          Location extraJarLocation = copyFileToLocation(extraJar, tempLocation);
```

Comment:
Seems like it's not a good idea to have `copyFileToLocation` to return `null`. All you need is in the `createPluginArchive`, you check if `pluginArchive` is null before calling `copyFileToLocation`.